apps/condo/domains/user/schema/SendMessageToSupportService.js

```diff
@@ -92,12 +92,23 @@ const SendMessageToSupportService = new GQLCustomSchema('SendMessageToSupportSer
 
                 const residents = await Resident.getAll(context, { user: { id: user.id } })
 
-                const organizationsIds = residents.reduce((accumulatedData, resident) => {
+                const organizationsIds = []
+                const residentsExtraInfo = []
+                for (let resident of residents) {
                     if (resident.organization && resident.organization.id) {
-                        return [...accumulatedData, resident.organization.id]
+                        organizationsIds.push(resident.organization.id)
                     }
-                    return accumulatedData
-                }, [])
+
+                    const residentInfo = { address: resident.address, accountNumber: null, paymentCategory: null }
+
+                    const serviceConsumer = await ServiceConsumer.getOne(context, { resident: { id: resident.id } })
```

Comment:
you can get all service consumers in one request before for loop

Reply:
It seems that 1 resident cannot have more than 1 ServiceConsumer. Or am I wrong?

Reply:
fixed